java-sdk/radar-schemas-tools/src/main/java/org/radarcns/schema/registration/SchemaTopicManager.java

```diff
@@ -0,0 +1,506 @@
+        if (storeTopic != null) {
+            try {
+                if (storeTopic.startsAtFirstId()) {
+                    storage.store(storeTopic);
+                } else {
+                    storage.storeInvalid(storeTopic);
+                }
+            } catch (IOException e) {
+                logger.error("Failed to store _schemas data", e);
+                throw e;
+            }
+        }
+    }
+
+    @NotNull
+    private Map<String, Object> getConsumerProps() {
+        Map<String, Object> consumerProps = new HashMap<>();
+
+        consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG,
+                "schema-backup-" + UUID.randomUUID().toString());
+        consumerProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "schema-backup");
+
+        consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, topics.getBootstrapServers());
+        consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        consumerProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+        consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+        consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+        return consumerProps;
+    }
+
+    @NotNull
+    private Map<String, Object> getProducerProps() {
+        Map<String, Object> producerProps = new HashMap<>();
+        producerProps.put(ProducerConfig.CLIENT_ID_CONFIG, "schema-backup");
+
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, topics.getBootstrapServers());
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                org.apache.kafka.common.serialization.ByteArraySerializer.class);
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                org.apache.kafka.common.serialization.ByteArraySerializer.class);
+        return producerProps;
+    }
+
+    /**
+     * Checks if the current _schemas topic is valid. If not, the topic is replaced by the data
+     * in the backup. The _schemas topic is considered valid if it exists, is not empty and starts
+     * at id 1.
+     * @param timeout time to wait for first schema records to become available.
+     * @throws SerializationException if a message in the topic cannot be read or written
+     * @throws ExecutionException if the topic configuration cannot be read or written
+     * @throws InterruptedException if the process was interrupted before finishing
+     * @throws RuntimeException storage failure or any other error.
+     * @throws IllegalStateException if this manager was not initialized or schema registry was
+     *                               running
+     * @throws KeeperException if no connection with Zookeeper could be made.
+     */
+    public void ensure(short replication, Duration timeout)
```

Comment:
FYI: entrypoint to check `_schemas` topic validity and replace it with a backup if needed.